includes/class-llms-awards-query.php

```diff
@@ -0,0 +1,525 @@
+<?php
+/**
+ * LLMS_Awards_Query class file.
+ *
+ * @package LifterLMS/Classes
+ *
+ * @since [version]
+ * @version [version]
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Query awarded achievements and engagements.
+ *
+ * @since [version]
+ */
+class LLMS_Awards_Query {
+
+	/**
+	 * Award type.
+	 *
+	 * Either "achievements" or "certificates".
+	 *
+	 * @var string
+	 */
+	public $type = '';
+
+	/**
+	 * User-submitted query args before sanitization, parsing, etc...
+	 *
+	 * @var array
+	 */
+	public $args = array();
+
+	/**
+	 * Final arguments passed to the WP_Query.
+	 *
+	 * @var array
+	 */
+	public $query_args = array();
+
+	/**
+	 * WP_Query object for the query instance.
+	 *
+	 * @var null
+	 */
+	public $query = null;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since [version]
+	 *
+	 * @param string $type       Award type to query. Accepts: "achievements" or "certificates".
+	 * @param array  $args {
+	 *     Query arguments.
+	 *
+	 *     @type array     $sort                   Array of sorting arguments where the key is value to sort by and the value is the direction.
+	 *     @type int|int[] $users                  Include awards for the specified WP_User(s) by WP_User ID.
+	 *     @type int|int[] $users__exclude         Exclude awards for the specified WP_User(s) by WP_User ID.
+	 *     @type int|int[] $related_posts          Include awards related to the specified WP_Post(s) by WP_Post ID.
+	 *     @type int|int[] $related_posts__exclude Exclude awards related to the specified WP_Post(s) by WP_Post ID.
+	 *     @type int|int[] $engagements            Include awards created from the specified `llms_engagement` post(s) by WP_Post ID.
+	 *     @type int|int[] $engagements__exclude   Exclude awards created from the specified `llms_engagement` post(s) by WP_Post ID.
+	 *     @type int|int[] $templates              Include awards created from the specified `llms_achievement` or `llms_certificate` template post(s) by WP_Post ID.
+	 *     @type int|int[] $templates__exclude     Exclude awards created from the specified `llms_achievement` or `llms_certificate` template  post(s) by WP_Post ID.
+	 *     @type boolean   $manual                 Include only awards created manually. If specified the `$related_posts`, `$related_posts__exclude`, `$engagements`, `$engagements__exclude`, `$templates`, and `$templates__exclude` arguments will be ignored.
+	 *     @type int       $page                   Results page number.
+	 *     @type int       $per_page               Number of results to display per page. Use `-1` to show all possible results.
```

Comment:
The actual WP_Query arg is `posts_per_page`.

Reply:
This isn't passed directly into WP_Query, but thank you.